packages/layer-composer/src/generators/heatmap/heatmap-animated.ts

```diff
@@ -88,6 +84,41 @@ const toURLArray = (paramName: string, arr: string[]) => {
     .join('&')
 }
 
+const getSublayersBreaks = (
+  config: GlobalHeatmapAnimatedGeneratorConfig,
+  intervalInDays: number
+) => {
+  // TODO - generate this using updated stats API ?
+  // TODO - in consequence, for each sublayer a different set of breaks should be produced
+  // TODO - BIVARIATE
+  if (config.combinationMode === 'bivariate') {
+    throw new Error('breaks generation not working for bivariate yet')
+  }
+  return config.sublayers.map(() => getBreaks(1, 30, 10, 1, [0.33, 0.66], intervalInDays))
+}
+
+const getLegends = (config: GlobalHeatmapAnimatedGeneratorConfig, intervalInDays: number) => {
+  const breaks = getSublayersBreaks(config, intervalInDays)
+  const ramps = getSublayersColorRamps(config)
+  return breaks.map((sublayerBreaks, sublayerIndex) => {
+    const ramp = ramps[sublayerIndex]
+    const legendRamp = sublayerBreaks.map((break_, breakIndex) => {
+      // TODO Omitting the Zero value hence the +1
+      const rampColor = ramp[breakIndex + 1] as string
+      const legendRampItem: [number, string] = [break_, rampColor]
+      return legendRampItem
+    })
+    const sublayerLegend: LayerMetadataLegend = {
```

Comment:
This needs to be defined out of the generator because will probably come from the dataview + dataset config... and I have two options in mind:
1. Prepare the data in `selectWorkspaceDataviewsResolved` and pass it down to the generator even it won't use it but seems easier as it contains all the legend data
2. Pass the dataviewInstance id to the generator and get it back to merge them in the app as you did [here](https://github.com/GlobalFishingWatch/frontend/pull/120/files#diff-1970e0dbbe9781cbaeda038430e2e38eR100)

I think the first one is easier but what are your thoughts ?

Reply:
Good catch, and definitely 2., which was easy to do